action/src/main.ts

```diff
@@ -1,26 +1,52 @@
-import * as core from '@actions/core'
-import { wait } from './wait'
+import * as core from '@actions/core';
+import { checkoutRepository } from './checkoutRepository';
+import { context } from '@actions/github';
+import getConfig from './getConfig';
+import { handlePullRequestRun } from './handlePullRequestRun';
+import { handlePushRun } from './handlePushRun';
+import { handleValidate } from './handleValidate';
+import { handleWriteActionSummary } from './handleWriteActionSummary';
+import { uploadCodeScan } from './uploadCodeScan';
+
+export enum RunStrings {
+  ValidationFailed = 'Validation failure. cfn-guard found violations.',
+  Error = 'Action failed with error'
+}
 
 /**
  * The main function for the action.
  * @returns {Promise<void>} Resolves when the action is complete.
  */
 export async function run(): Promise<void> {
-  try {
-    const ms: string = core.getInput('milliseconds')
+  const { analyze, checkout } = getConfig();
+  const { eventName } = context;
 
-    // Debug logs are only output if the `ACTIONS_STEP_DEBUG` secret is true
-    core.debug(`Waiting ${ms} milliseconds ...`)
+  checkout && (await checkoutRepository());
 
-    // Log the current timestamp, wait, then log the new timestamp
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+  try {
+    const result = await handleValidate();
+    const {
+      runs: [sarifRun]
+    } = result;
 
-    // Set outputs for other workflow steps to use
-    core.setOutput('time', new Date().toTimeString())
+    if (sarifRun.results.length) {
+      if (analyze) {
+        core.setFailed(RunStrings.ValidationFailed);
+        await uploadCodeScan({ result });
+      } else {
+        const results =
+          eventName === 'pull_request'
```

Comment:
nit: not sure if this type will used more but let's avoid the hardcode of 'pull-request' freeze it or enum it

Reply:
yeah - ill make a shared enum for event names good call